components/cli/src/password.rs

```diff
@@ -0,0 +1,235 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This module contains utilities for passphrase entry.
+
+use failure::{bail, Error, ResultExt};
+use rpassword::read_password_from_tty;
+use serde::{Deserialize, Serialize};
+use zeroize::Zeroize;
+
+use std::ops::{Deref, DerefMut};
+use std::{env, str::FromStr};
+
+/// Default name of the environment variable with a consensus key passphrase.
+pub const DEFAULT_CONSENSUS_PASS_ENV_VAR: &str = "EXONUM_CONSENSUS_PASS";
+/// Default name of the environment variable with a service key passphrase.
+pub const DEFAULT_SERVICE_PASS_ENV_VAR: &str = "EXONUM_SERVICE_PASS";
+
+/// A wrapper around `String` which securely erases itself on drop.
+#[derive(Debug, Default, Serialize, Deserialize, PartialEq)]
+pub struct Passphrase(pub String);
+
+impl Drop for Passphrase {
+    fn drop(&mut self) {
+        self.0.zeroize()
+    }
+}
+
+impl Deref for Passphrase {
+    type Target = String;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl DerefMut for Passphrase {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+impl Passphrase {
+    /// Read the passphrase from stdin.
+    pub fn read_from_tty(prompt: &str) -> Result<Self, Error> {
+        Ok(Self(read_password_from_tty(Some(prompt))?))
+    }
+}
+
+/// Passphrase input method.
+///
+/// Defaults to `Terminal`.
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub enum PassInputMethod {
+    /// Prompt passphrase from terminal.
+    Terminal,
+    /// Get passphrase from the environment variable with given name.
+    ///
+    /// Default values are `EXONUM_CONSENSUS_PASS` and `EXONUM_SERVICE_PASS` for
+    /// consensus and service secret keys correspondingly.
+    /// Defaults are used if `None` is provided.
+    EnvVariable(Option<String>),
+    /// Passphrase is passed as a command line parameter.
+    CmdLineParameter(Passphrase),
+}
+
+impl Default for PassInputMethod {
+    fn default() -> Self {
+        PassInputMethod::Terminal
+    }
+}
+
+/// Secret key types.
+#[derive(Copy, Clone)]
+pub enum SecretKeyType {
+    /// Consensus key. Used in communication between nodes during consensus.
+    Consensus,
+    /// Service key. Used to sign transactions produced by the node.
+    Service,
+}
+
+impl SecretKeyType {
+    /// Returns default environment variable names for the corresponding `SecretKeyType`.
```

Comment:
@Maria-Nosyk suggests to use infinitives for methods documentation.

Reply:
I suggest that we develop a style guide and then bring everything to an organized state. `-s` form looks more natural here as for me. So, as we plan to edit this in the future, you can leave it as is

Reply:
Discussed privately with Maria, I will abstain from any changes before the release of the official style guide.